resources/aws_ec2.go

```diff
@@ -912,6 +800,84 @@ func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
 
+// describeInstanceByName takes an ec2 client session and an instance name, and
+// returns an ec2Instance or an error.
+func describeInstanceByName(c *ec2.EC2, name string) (ec2Instance, error) {
```

Comment:
the biggest unknown to me, is why is this a special type we made (maybe it's useful, you have to convince me) instead of just plainly using https://godoc.org/github.com/aws/aws-sdk-go/service/ec2#Instance ? (or possibly a pointer to that actually)

Reply:
I could do it with ec2.Instance, but here's why I chose to do this:
- I think it's cleaner to use instance.ID and instance.State vs. instance.InstanceId and instance.State.Name
- ec2Instance only contains information we care about, and when we add things like IP addresses, it will be more clear what's going on, IMO.
We talked about wrapping their api in our own (saner) api, and I think this is part of it. If you disagree, I'm happy to change it. :)

Reply:
> I think it's cleaner to use instance.ID and instance.State vs. instance.InstanceId and instance.State.Name

Why? You've arguably just copied that information, left out a bunch of potentially useful things, and removed one dot. If the argument was that we added additional information to our ec2Instance struct that needed to be cached or stored, I'd be convinced, but a straight copy doesn't do it for me.

In that second case where we need to store additional things, I'd probably even recommend having out ec2Instance be a:

```
type ec2Instance struct {
        *ec2.Instance // anonymous struct field, look it up

        someSpecialField string
        someOtherField uint32
}
```

> it will be more clear what's going on, IMO.

You haven't said _why_.

> We talked about wrapping their api in our own (saner) api

Yes, our _external_ API. The struct fields and behaviours that we expose to the end user is what should be sane, even if it differs from what AWS's public API does. What we're discussing here is our internal helper functions.